### ISSUE TITLE
Integrating Quack.RMSNorm into real training codebase - fp32 for weights is not realistic (only for reference testing?), needs to accept 3D inputs

### ISSUE THREAD

**Issue by lessw2020** [Dao-AILab/quack#12]
I'm currently integrating the quack.rmsnorm kernel into TorchTitan to showcase these impressive kernels.

However, during integration I hit two key issues;
a -  RMSNorm has a hard assert that the weights are fp32?  This seems unrealistic for real world use...bf16 works nicely and is the default for most. 
b - RMSNorm also has a hard assert on 2D inputs...but most inputs for actual training are 3D.  This one is not too hard to resolve but I think it would be good to integrate this into the kernel wrapper to resolve.

For fp32 - I wanted to check if there is some other reason I'm not aware of why fp32 is enforced (assuming it may just be b/c much easer to verify results vs reference, using fp32 for testing, but now that it's moving to real world use can we lift this?)

**Comment by tridao** [comment]
No real reason why weight should be in fp32, I usually use torch AMP for small models where rmsnorm weights are fp32. Should be easy to change. The only place that needs attention is that we still want to compute the partial weight gradient in fp32 before summing up, then convert that to bf16. I might take a look a bit later.
3D input should also be easy to change, just a reshape at the beginning and end of the function (in pytorch).

**Comment by lessw2020** [comment]
Thanks @tridao  - I've made a PR to handle the 3D input portion.
https://github.com/Dao-AILab/quack/pull/13

Would be great if you can update the kernel for bf16 support.   Otherwise, I'll take a crack at it tomorrow paying attention to your comments above.

**Comment by lessw2020** [comment]
I've made a PR (https://github.com/Dao-AILab/quack/pull/16) that updates support for weights in BF16.  
Let me go ahead and also add in support for fp16 as well for completeness!

**Comment by lessw2020** [comment]
We are now up and running in TorchTian via draft pr, and both issues above have been resolved via landed quack prs...thus closing this tracking issue.
